### ISSUE TITLE
Calling Subscriptions from Clojure REPL

### ISSUE THREAD

**Issue by Wolfgang51** [day8/re-frame#522]
After doing some research I found that most of re-frame is now in `.cljc` (https://github.com/Day8/re-frame/blob/master/docs/FAQs/Why-CLJC.md). This was excited to hear this due to the fact that we can now test from the JVM now and not have to rely on a JV environment. However, when I tried exercising _subscriptions_ in the REPL I came to the following issues.

**1. Calling subscriptions that subscribe to other subscriptions and use their values for computations lead to an error message of invalid arity.** 

--- Subscription ---
```clj
; Subscription
(re-frame/reg-sub
 ::suggestions
 (fn [db [_ query]]
   (get-in db [:typeahead :suggestions])))

(re-frame/reg-sub
 ::query
 (fn [db [_ query]]
   (get-in db [:typeahead :query])))

(re-frame/reg-sub
 ::typeahead
 (fn [_ _]
   [(re-frame/subscribe [::query])
    (re-frame/subscribe [::suggestions])])
 (fn [[query suggestions] _]
   {:query query
    :suggestions suggestions}))
``` 
-- Error --
```
slm.components.typeahead.subscriptions=> (re-frame/subscribe [::typeahead])

Execution error (ArityException) at re-frame.subs/reg-sub$subs-handler-fn (subs.cljc:325).
Wrong number of args (1) passed to: slm.components.typeahead.subscriptions/eval28820/fn--28821

(let [typeahead @(re-frame/subscribe [::typeahead]                                         )]
                                    #_=>   typeahead)

Execution error (ArityException) at re-frame.subs/reg-sub$subs-handler-fn (subs.cljc:325).
Wrong number of args (1) passed to: slm.components.typeahead.subscriptions/eval27322/fn--27323
```

**2. Can't extract values from subscriptions** 

```clj
; db
(def default-db
  {:name "olanext"})

; subscription
(re-frame/reg-sub
 ::name
 (fn [db _]
   (:name db)))
```
--- attempt to extract value ---
```
slm.components.typeahead.subscriptions=> (let [name @(re-frame/subscribe [::name])]
                                    #_=>   name)
nil
```

Is there any resolution to these issues?

**Comment by dawran6** [comment]
> 2. Can't extract values from subscriptions

I think this is because you haven't put any data into the app-db yet. You can do this by running this in the REPL:

```clojure
(re-frame/reg-event-db
 :initialize-db
 (fn [_ _] default-db))

(re-frame/dispatch [:initialize-db])
```

**Comment by dawran6** [comment]
> 1. Calling subscriptions that subscribe to other subscriptions and use their values for computations lead to an error message of invalid arity.

I can reproduce the exception. I traced the exception to this [line](https://github.com/Day8/re-frame/blob/master/src/re_frame/subs.cljc#L368):

```clojure
    (register-handler
      kind
      query-id
      (fn subs-handler-fn
        ([db query-vec]
         (let [subscriptions (inputs-fn query-vec)  ;; HERE
```

When the error happened, the `query-vec` was `[:user/typeahead]` and the `inputs-fn` was what's defined in the `::typeahead` subscription:

```clojure
(fn [_ _]
   [(re-frame/subscribe [::query])
    (re-frame/subscribe [::suggestions])])
```
I'm not sure why normal cases (when not tested in the REPL) we do not see this issue, and I'll try to figure it out. It'll be great to have some guidance from the maintainers.

**Comment by dawran6** [comment]
Hi maintainers, I think the actions for this issue is to improve documentation. We can be clearer about writing signal functions and how to use layer 3 subscription in general.

**Comment by superstructor** [comment]
Fixed in #541
